packages/debug-nodejs/src/node/node-debug-adapter-contribution.ts

```diff
@@ -20,19 +20,41 @@ import { injectable } from 'inversify';
 // tslint:disable-next-line:no-implicit-dependencies
 import { FileUri } from '@theia/core/lib/node';
 import { DebugConfiguration } from '@theia/debug/lib/common/debug-configuration';
-import { AbstractVSCodeDebugAdapterContribution } from '@theia/debug/lib/node/vscode/vscode-debug-adapter-contribution';
+import { VSCodeDebugAdapterContribution } from '@theia/debug/lib/node/vscode/vscode-debug-adapter-contribution';
+import { DebugAdapterContribution, DebugAdapterExecutable } from '@theia/debug/lib/node/debug-model';
+import { IJSONSchema, IJSONSchemaSnippet } from '@theia/core/lib/common/json-schema';
 
 export const INSPECTOR_PORT_DEFAULT = 9229;
 export const LEGACY_PORT_DEFAULT = 5858;
 
 @injectable()
-export class NodeDebugAdapterContribution extends AbstractVSCodeDebugAdapterContribution {
+export class NodeDebugAdapterContribution implements DebugAdapterContribution {
```

Comment:
Can you elaborate why it cannot extend `VSCodeDebugAdapterContribution`? the same for node2 and java

Reply:
In case of extension I have to use inversify. 
But in this case I can't use it in plugin-ext

Reply:
I think `VsCodePluginScanner` should be improved to handle vscode extension localization generally and support debuggers (as well Theia plugin namespace?). The plugin extension then should use `PluginModel` to provide `DebugAdapterContribution`. Maybe some code should be copied there from `VSCodeDebugAdapterContribution` to achieve it.

Reply:
@akosyakov resolved
https://github.com/theia-ide/theia/pull/3804/commits/2aa725a8cdc679e3d62facceb0fc4880fa7b2978